ttnn/cpp/ttnn/operations/ccl/all_gather/device/multi_core/all_gather_op_multi_core.cpp

```diff
@@ -938,6 +980,29 @@ operation::ProgramWithCallbacks all_gather_multi_core_with_workers(const Tensor&
                             emit_sharded_tensor_kernel_rt_args(device, output_tensor, worker_writer_receiver_rt_args);
                         }
 
+                        /* Overlap params */
+                        if (overlap_op) {
+                            // Push the receiver worker core noc coordinates
+                            for (auto& noc_coord : receiver_worker_cores_noc_coords) { // Should run global_num_workers times
+                                worker_writer_receiver_rt_args.push_back(static_cast<uint32_t>(noc_coord.x));
+                                worker_writer_receiver_rt_args.push_back(static_cast<uint32_t>(noc_coord.y));
+                            }
+
+                            if (datacopy_cores.has_value()) {
```

Comment:
Please rename all instances of `datacopy_cores` with something like `fused_receiver_op_cores`

Reply:
Also to simplify porting, can we move this logic to a new file (`ccl_op_fusion` or something like that)? It'll simplify adding that type of support to future CCLs